filament/backend/src/vulkan/VulkanHandles.cpp

```diff
@@ -352,11 +352,17 @@ VulkanRenderTarget::VulkanRenderTarget(VkDevice device, VkPhysicalDevice physica
         if (texture && texture->samples == 1) {
             auto msTexture = texture->getSidecar();
             if (UTILS_UNLIKELY(!msTexture)) {
+                // Clear all usage flags that are not related to attachments, so that we can
+                // use the transient usage flag.
+                const TextureUsage usage = texture->usage & TextureUsage::ALL_ATTACHMENTS;
```

Comment:
Is this correct? what if there was some other usage bits set? Shouldn't we instead passing the `usage` as is and let the VulkanTexture decide if it can use it?

Reply:
As far as I understand, `texture->usage` is the `usage` of the resolved texture.
We cannot use that for MSAA, as the requirement are not as strict.

In this particular case, the `SAMPLEABLE` flag is raised for the original texture, but it is not necessary nor desirable for the MSAA texture, and it prevents us from setting the attachment as transient. We cannot have any flags besides the four ***_ATTACHMENT_BIT for transient attachments.

Reply:
Thinking about it... we could eventually remove the `preferTransientAttachment` parameter from the `VulkanTexture`. I don't think there is a case where you would not want to create a transient attachment if it is possible to do so.
What do you think?

Reply:
> As far as I understand, `texture->usage` is the `usage` of the resolved texture. We cannot use that for MSAA, as the requirements are stricter for transient attachments.
> 
> In this particular case, the `SAMPLEABLE` flag is raised for the original texture, but it is not necessary nor desirable for the MSAA texture, and it prevents us from setting the attachment as transient. We cannot have any flags besides the four ***_ATTACHMENT_BIT for transient attachments.

mh... I want to understand why the SAMPLEABLE flag is set then. Did the framegraph set it? if it did, it's probably that it needs it?

Reply:
I don't know the frame graph well enough to confirm that. I am not sure if that helps, but by the time `ResourceAllocator::createTexture` is called, the `SAMPLEABLE` flag is already there.

However, I think it is not relevant for MSAA. Whether the resolved texture is sampled or not should not impact the MSAA texture usage in any way. But, we still need to use the right attachment flags, hence the mask with `TextureUsage::ALL_ATTACHMENTS`.

Reply:
As discussed offline, I checked the way filament handles the MSAA resolve.
The `VK_IMAGE_USAGE_TRANSFER_SRC_BIT` flag is only needed when using `vkCmdResolveImage`. Filament uses the more efficient `VkSubpassDescription::pResolveAttachments` for the sidecar textures, so we are safe masking the flags unrelated to attachments.